pkgs/by-name/ta/tabby-terminal/package.nix

```diff
@@ -0,0 +1,215 @@
+    patches = [ ./fix-argv-prefix-splice.patch ];
+
+    electronVersion = "29.4.6";
+
+    electronHeaders = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronVersion}/node-v${electronVersion}-headers.tar.gz";
+      hash = "sha256-e0qiEGI/1rWK72oLXfKtbbc8KmOGMhMx4IbT6MGAjms=";
+    };
+
+    electronHeadersSHA = fetchurl {
+      url = "https://www.electronjs.org/headers/v${electronVersion}/SHASUMS256.txt";
+      hash = "sha256-0mCnKIR3GS1TL+g0h+y2cAlqbBogf88WxN8xJvmFl4g=";
+    };
+
+    buildInputs = [
+      nodejs
+      python3
+      fontconfig
+      electron_29
+    ];
+
+    nativeBuildInputs = [
+      git
+      yarn
+      patch-package
+      prefetch-yarn-deps
+      fixup-yarn-lock
+      (python3.withPackages (python-pkgs: [
+        python-pkgs.distutils
+        python-pkgs.gyp
+      ]))
+      makeWrapper
+      http-server
+      pkg-config
+      libsecret
+      desktop-file-utils
+    ];
+
+    configurePhase =
+      ''
+        export buildDir=$PWD
+      ''
+      # Loop through all the yarn caches and install the deps for the respective package
+      + builtins.concatStringsSep "\n" (
+        map (cache: ''
+          cd ${cache.pkg}
+          export HOME=$PWD/yarn_home
+          fixup-yarn-lock yarn.lock
+          yarn config --offline set yarn-offline-mirror ${cache.cache}
+          echo "Installing deps for ${cache.pkg}"
+          yarn install --offline --prefer-offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+          patch-package
+          patchShebangs node_modules
+          echo "Done!"
+          cd $buildDir
+        '') yarnCaches
+      )
+      + ''
+        cd $buildDir/node_modules
+      ''
+      # Loop thought the "built in" plugins and link them to the node_modules
+      + builtins.concatStringsSep "\n" (
+        map (plugin: ''
+          ln -fs ../${plugin} ${plugin}
+        '') builtinPlugins
+      )
+      + ''
+        cd $buildDir
+      '';
+
+    buildPhase =
+      # Start a fake (and completely unnecessary) http-server to let electron-rebuild "download" the headers and hashes
+      ''
+        mkdir -p http-cache/v${electronVersion}
+        cp $electronHeaders http-cache/v${electronVersion}/node-v${electronVersion}-headers.tar.gz
+        cp $electronHeadersSHA http-cache/v${electronVersion}/SHASUMS256.txt
+        http-server http-cache &
+      ''
+      # Rebuild all the needed packages using electron-rebuild
+      + builtins.concatStringsSep "\n" (
+        map (pkg: ''
+          yarn --offline electron-rebuild -v ${electronVersion} -m ${pkg} -f -d "http://localhost:8080"
+        '') rebuildPkgs
+      )
+      + ''
+          kill $!
+          yarn build
+
+          mkdir -p $out/share/tabby
+          mkdir -p $out/bin
+          cp -r . $out/share/tabby
+          makeWrapper "${electron_29}/bin/electron" "$out/bin/tabby" --add-flags "$out/share/tabby/app" --set TABBY_DEV 1
```

Comment:
Why does `TABBY_DEV` need to be set?

Reply:
The code checks for TABBY_DEV, and alters a few import paths based on its value. Their default values are set to something that would exist with electron-builder, which we are not using.

Reply:
Might be best to document that then :)